android/KMAPro/build.sh

```diff
@@ -44,97 +80,62 @@ function makeLocalSentryRelease() {
-
-KEYBOARD_PACKAGE_ID="sil_euro_latin"
-KEYBOARDS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${KEYBOARD_PACKAGE_ID}.kmp"
-MODEL_PACKAGE_ID="nrc.en.mtnt"
-MODELS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${MODEL_PACKAGE_ID}.model.kmp"
-
-# Verify default keyboard and dictionary exist
-if [[ ! -f "$KEYBOARDS_TARGET" ]]; then
-  echo "$KEYBOARDS_TARGET doesn't exist. Will download the latest version"
-  DO_KEYBOARDS_DOWNLOAD=true
-fi
+#### Build action definitions ####
 
-if [[ ! -f "$MODELS_TARGET" ]]; then
-  echo "$MODELS_TARGET doesn't exist. Will download the latest version"
-  DO_MODELS_DOWNLOAD=true
-fi
+# Check about cleaning artifact paths and upload directories
+if builder_start_action clean; then
+  cd "$KEYMAN_ROOT/android/KMAPro/"
 
-# Local development optimization to upload local symbols to Sentry
-if [[ $VERSION_ENVIRONMENT == "local" && $ONLY_DEBUG == true && $UPLOAD_SENTRY == true ]]; then
-  DO_SENTRY_LOCAL_UPLOAD=true
-fi
+  if [ -d "$KEYMAN_ROOT/android/KMAPro/kMAPro/build/outputs" ]; then
+    rm -rf "$KEYMAN_ROOT/android/KMAPro/kMAPro/build/outputs"
+  fi
 
-echo
-echo "NO_DAEMON: $NO_DAEMON"
-echo "ONLY_DEBUG: $ONLY_DEBUG"
-echo "DO_KEYBOARDS_DOWNLOAD: $DO_KEYBOARDS_DOWNLOAD"
-echo "DO_MODELS_DOWNLOAD: $DO_MODELS_DOWNLOAD"
-echo "DO_SENTRY_LOCAL_UPLOAD: $DO_SENTRY_LOCAL_UPLOAD"
-echo
+  if [ -d "$KEYMAN_ROOT/android/upload" ]; then
+    rm -rf "$KEYMAN_ROOT/android/upload"
+  fi
 
-if [ "$NO_DAEMON" = true ]; then
-  DAEMON_FLAG=--no-daemon
-else
-  DAEMON_FLAG=
+  builder_finish_action success clean
 fi
 
-# Convert markdown to html for offline help
-echo "Converting markdown to html for offline help"
-cd "$KEYMAN_ROOT/android"
-./build-help.sh htm
-cd "$KEYMAN_ROOT/android/KMAPro"
+if builder_start_action configure; then
+  cd "$KEYMAN_ROOT/android/KMAPro/"
 
-# Download default keyboard and dictionary
-if [ "$DO_KEYBOARDS_DOWNLOAD" = true ]; then
-  downloadKeyboardPackage "$KEYBOARD_PACKAGE_ID" "$KEYBOARDS_TARGET"
-fi
+  # Copy Keyman Engine for Android
+  cp "$KEYMAN_ROOT/android/KMEA/app/build/outputs/aar/${CONFIG}/keyman-engine.aar" "$KEYMAN_ROOT/android/KMAPro/kMAPro/libs/keyman-engine.aar"
+
+  KEYBOARD_PACKAGE_ID="sil_euro_latin"
+  KEYBOARDS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${KEYBOARD_PACKAGE_ID}.kmp"
+  MODEL_PACKAGE_ID="nrc.en.mtnt"
+  MODELS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${MODEL_PACKAGE_ID}.model.kmp"
 
-if [ "$DO_MODELS_DOWNLOAD" = true ]; then
+  downloadKeyboardPackage "$KEYBOARD_PACKAGE_ID" "$KEYBOARDS_TARGET"
   downloadModelPackage "$MODEL_PACKAGE_ID" "$MODELS_TARGET"
+
+  builder_finish_action success configure
 fi
 
-if [ "$ONLY_DEBUG" = true ]; then
-  BUILD_FLAGS="assembleDebug lintDebug"
-else
-  # build = assemble + check; check = test + lint
-  BUILD_FLAGS=build
+if builder_start_action build; then
+
+  # Convert markdown to html for offline help
+  _convert_markdown_to_html
+
+  cd "$KEYMAN_ROOT/android/KMAPro"
```

Comment:
This folder restoration should be done in the `convert_markdown_to_html` function